### DISCUSSION TITLE
Where can I find all .env variables related to PrivateBin?

### DISCUSSION THREAD

**Discussion by uhlhosting** [PrivateBin/PrivateBin#1066] [Q&A]
I would like to see a list of all env variables and could not find one.

**Comment by elrido** [comment]
[searching for the PHP `genenv` function](https://github.com/PrivateBin/PrivateBin/search?q=getenv) reveals only two uses:
- `CONFIG_PATH` allows changing the path where the configuration file is read from - it is [mentioned in one FAQ entry](https://github.com/PrivateBin/PrivateBin/wiki/FAQ#how-can-i-link-to-a-read-only-mode-where-users-cant-submit-pastes), explaining the use case for it
- `PRIVATEBIN_GCS_BUCKET` seems related to the Google Cloud Storage backend

In the container image's [PHP-fpm configuration](https://github.com/PrivateBin/docker-nginx-fpm-alpine/blob/087acc8b5619a3d5d56450c1671b13cab930d6a9/etc/php81/php-fpm.d/zz-docker.conf#L15-L19), there are additional environment variables mentioned, that get passed through to PHP:
```
env[GOOGLE_APPLICATION_CREDENTIALS] = $GOOGLE_APPLICATION_CREDENTIALS
env[GOOGLE_CLOUD_PROJECT] = $GOOGLE_CLOUD_PROJECT
env[GCLOUD_PROJECT] = $GCLOUD_PROJECT
```
I assume it's safe to guess these are all related to the Google Cloud Storage backend.

**Reply by uhlhosting** [reply]
Which is correct from the two?

<img width="588" alt="image" src="https://user-images.githubusercontent.com/6324047/221150018-50b38024-b249-4658-9d70-589608b657bc.png">

I was sure there are more env variables, maybe not.

**Reply by rugk** [reply]
> I was sure there are more env variables, maybe not.

No, configuration is mostly done via an ini file: https://github.com/PrivateBin/PrivateBin/wiki/Configuration

I would guess the timezone variables you have there do not matter.

**Reply by elrido** [reply]
Ah, I had understood you were talking about PrivateBin related environment variables. Those two are relevant for the logging performed by nginx & php-fpm in the container image, for example. They are mentioned here:
https://github.com/PrivateBin/docker-nginx-fpm-alpine/#timezone-settings

> Which is correct from the two?

AFAIK both are required: TZ influences nginx and for some reason php-fpm had to use their own PHP_TZ. Maybe share-hosters have the scenario where they have the host running on one timezone setting, i.e. based on it's physical location, while still allowing for a per-customer timezone setting for PHP?

**Reply by uhlhosting** [reply]
Yes i use docker. Forget to mention.
